### PR TITLE
JAPI-230: final axis2 refactoring for salt/hpcc use of wsclient

### PR DESCRIPTION
**Pull request by drealeed**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java [hunk 1]
```diff
@@ -325,20 +329,46 @@ static public DelimitedDataOptions createDelimitedDataOptionsObject(String recor
...
             {
-                dropZonesWrapper[i] = new DropZoneWrapper(dropZone[i]);
+                dropZonesWrapper.add(new DropZoneWrapper(dropZone[i]));
             }
         }
 
         return dropZonesWrapper;
     }
 
+    public String copyFile(String from,String to, boolean overwrite) throws Exception, ArrayOfEspExceptionWrapper {
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java]: We should be adding junits for all new helper methods...

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java [hunk 2]
```diff
@@ -325,20 +329,46 @@ static public DelimitedDataOptions createDelimitedDataOptionsObject(String recor
...
+        verifyStub();
+        Copy cp=new Copy();
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java]: very minor, just to be consistent let's rename cp as request

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 3]
```diff
@@ -69,6 +71,7 @@
 import org.w3c.dom.Document;
 import org.w3c.dom.NodeList;
 
+
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: let's remove

**Comment by drealeed** [comment]
I'll create some integration level/live tests..or do you have some sort of powermock framework you use for mocking up tests like these for running as unit tests?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java (hunk 1 above)
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java]: minor format issue - move curly to new line

**Comment by rpastrana** [comment]
> 
> 
> I'll create some integration level/live tests..or do you have some sort of powermock framework you use for mocking up tests like these for running as unit tests?

not yet, our jenkins nightly build launch an HPCC instance on localhost, but it's currently empty, so we'd have to create the data/files as part of the junit. We will soon have pre-caned data to test against...

**Pull request merged**